packages/experiment-tag/src/experiment.ts

```diff
@@ -314,13 +318,14 @@ export const setUrlChangeListener = () => {
 
     // Wrapper for replaceState
     history.replaceState = function (...args) {
-      previousUrl = globalScope.location.href;
```

Comment:
Can you explain this change to me? By moving this to below the apply, we've changed the behavior of the redirect logic in applyRedirect. Since previousUrl will not have been updated.

Reply:
The previous logic was inaccurate - it should've been updated afterwards.